test/support/pages/SettingsPage.js

```diff
@@ -0,0 +1,272 @@
+// in test/support/pages/SettingsPage.js
+define(function (require) {
+  // the page object is created as a constructor
+  // so we can provide the remote Command object
+  // at runtime
+
+  var Common = require('./Common');
+
+  var defaultTimeout = 5000;
+  var common;
+
+  function SettingsPage(remote) {
+    this.remote = remote;
+    common = new Common(this.remote);
+  }
+
+  SettingsPage.prototype = {
+    constructor: SettingsPage,
+
+    getTimeBasedEventsCheckbox: function getTimeBasedEventsCheckbox() {
```

Comment:
@LeeDr I think in the case of methods added to a function's prototype, the repetition is unnecessary as long as the constructor function is named. For instance I removed the name from the `createIndex` method, and I'm still getting an intelligible stack trace:

`at SettingsPage.createIndex  </Users/matt/Code/kibana/test/support/pages/SettingsPage.js:235:7>`

I only worry about the repetition because it creates two places the name needs to be updated if someone makes a change, and if they get out of sync it could be very confusing. Using the function declaration syntax as the expression for a variable assignment also has some confusing effects on scoping: http://stackoverflow.com/a/338053